### ISSUE TITLE
Connection to the notebook server could not be established on WebFaction via https with pw

### ISSUE THREAD

**Issue by JoeCodeswell** [jupyter/help#98]
Thanks for a GREAT App.

I am running jupyter notebook --version 4.2.3 on WebFaction via SSl with a password.

The tree comes up just fine & i can see sub directories in it. 

But  when i try to execute a cell, from Chrome on [windows 10 & Android 4 ] or FireFox on windows 10, I (eventually) get "a connection to the notebook server could not be established", on https://jups.joecodeswell.org.  Jupyter DID work in the past when the site was

- NOT SSL secured
- Not password protected

Attachment01.txt is some info from the nohup.out file. The "Replacing stale connection" messages seem to be a hint of what is wrong.

Thanks for the help in advance.

Love and peace,

Joe
[Attachment01.txt](https://github.com/jupyter/help/files/607074/Attachment01.txt)

**Comment by minrk** [comment]
Is the SSL cert self-signed, or otherwise not fully trusted by your browser? Safari is known to refuse to connect websockets in these cases, and it's possible that other browsers are following suit.

If that's the case, could you use [Let's Encrypt](https://letsencrypt.org) to generate a fully trusted cert?

**Comment by JoeCodeswell** [comment]
Dear Min,

Thanks for the response.  The "SSL cert" is signed & "fully trusted by" Chrome & FireFox. I used Let's Encrypt. 

Here's what happens. After I log in i can see directories and files. After i click on a notebook i get a screen that looks like the attached Clipboard02.jpg. Notice the orange "Connecting to kernel" message in the upper right. After a while i get a  "Connection failed" screen that looks like Clipboard03.jpg.

I am beginning to suspect that the problem is associated with the use of "session_id"s like 1F1CC7DB006540979C06CC49DCAFC85D. I have been staring at the nohup.out file attached as 20161125_8.36AM_nohup.out.txt. I have NO IDEA if i am on the right track or not :) 

Anyway, Min, thanks for the help in advance.

Love and peace,

Joe

Clipboard02.jpg
![clipboard02](https://cloud.githubusercontent.com/assets/3194021/20630950/fe58f5ac-b2e8-11e6-9644-0509814db80b.jpg)
Clipboard03.jpg
![clipboard03](https://cloud.githubusercontent.com/assets/3194021/20630952/0253c2c2-b2e9-11e6-9b2d-1b99f7efbde5.jpg)

[20161125_8.36AM_nohup.out.txt](https://github.com/jupyter/help/files/613638/20161125_8.36AM_nohup.out.txt)

**Comment by JoeCodeswell** [comment]
Dear Min,

Is there anything happening with this issue?

Thanks in advance.

Love and peace,
Joe

**Comment by minrk** [comment]
Sorry for the silence. Do you see anything in the js console when this happens? It's a bit odd that it appears to be getting started with the connection, but then dying (I believe this is the cause of the stale connection message). Are there any proxies in between the notebook server and your browser (e.g. AWS services, etc.)?

**Comment by Anwesha227** [comment]
Yes I have been facing a similar issue. There is a proxy in between, which I tried to bypass via Chrome.
I tried this route: Settings--> Advanced-->Open Proxy settings-->Connections-->LAN settings-> Advanced

Then under Exceptions (Do not use proxy server for addresses beginning with:) I entered the external IP over there according to the format followed. But this doesn't seem to work?
